### ISSUE TITLE
Add local "Bookmarks" list (not Twitter's "Bookmarks")

### ISSUE THREAD

**Issue by Symbian9** [IBBoard/cawbird#247]
**Is your feature request related to a problem? Please describe.**

Add feature to locally bookmark tweets & store local bookmark list in HTML file.

> NOTE: This feature request is **NOT about Twitter's "Bookmarks" feature** mentioned in https://github.com/IBBoard/cawbird/issues/76

**Describe the solution you'd like**

Add feature to create local bookmarks list in HTML format, same way & same format as web-page bookmarks works in Firefox & other browsers.

There should be few options:

- Add/Remove to/form local bookmarks list (via bookmark button in exact tweet contextual menu on RMB):
- Import local bookmarks from HTML;
- Export local bookmarks to HTML;

Local bookmarks storage: `~./cawbird/cawbird-bookmarks.html`

Add local bookmarks backup preferences to set period between backups, regularity & max number of stored backups.

**Describe alternatives you've considered**

**Additional context**

**Comment by IBBoard** [comment]
I'd suggest that we use the database rather than a HTML file by default, but it could be a useful feature for some people.

I think backups are out of scope, though. We don't back up the rest of the user credentials, DMs, etc.

**Comment by Symbian9** [comment]
> I'd suggest that we use the database rather than a HTML file by default, but it could be a useful feature for some people.

For Cawbird internal storage maybe database is good choice, but there should be at least "Export/Import local bookmarks (HTML)" feature too.

> I think backups are out of scope, though. We don't back up the rest of the user credentials, DMs, etc.

To be clear, I'm talking about backups of local bookmarks just after add new bookmark.

**Comment by IBBoard** [comment]
> For Cawbird internal storage maybe database is good choice, but there should be at least "Export/Import local bookmarks (HTML)" feature too.

Yes, would need an "export" option as well. But having the default store as HTML/XML is inefficient, and the export could go anywhere.

(Aside: `~/.cawbird/` would be a bad idea. Apps shouldn't create their own dot folders. It's annoying and non-standard)

> To be clear, I'm talking about backups of local bookmarks just after add new bookmark.

Sorry, still don't see where backups are relevant. If we're storing them in the database then we'd have a "bookmark" table with a row per tweet. When we add a bookmark we add a row. Nothing to back up. And when we export then we explicitly *shouldn't* back up because the user might not want that, and if they do then they can copy the file themselves.

**Comment by Symbian9** [comment]
> Nothing to back up.

What if Cawbird just would broke database file on the next bookmark adding? (by some Cawbird crash or by casual PC switch off)

So, at least one backup of database should be stored, "one bookmark behind" main database file.
